contracts/src/tokens/memecoin.cairo

```diff
@@ -216,44 +205,65 @@ mod UnruggableMemecoin {
         ///
         /// Note that when transfers are done, between addresses that already
         /// hold tokens, we do not increment the number of holders. it only
-        /// gets incremented when the recipient that hold no tokens
+        /// gets incremented when the recipient that hold no tokens.
+        /// But if the sender will no longer hold tokens after the transfer,
+        /// the number of holders is decremented.
         ///
         /// # Arguments
+        /// * `sender` - The sender of the tokens being transferred.
         /// * `recipient` - The recipient of the tokens being transferred.
+        /// * `amount` - The amount of tokens being transferred.
         #[inline(always)]
-        fn _enforce_holders_limit(ref self: ContractState, recipient: ContractAddress) {
-            // enforce max number of holders before launch
+        fn _enforce_holders_limit(
+            ref self: ContractState,
+            sender: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) {
+            if !self.launched.read() {
+                // if the sender will no longer hold tokens
+                if sender.is_non_zero() && self.balance_of(sender) == amount {
+                    let current_holders_count = self.pre_launch_holders_count.read();
+
+                    // decrease holders count
+                    self.pre_launch_holders_count.write(current_holders_count - 1);
+                }
+
+                // if the recipient doesn't hold tokens yet
+                if self.balance_of(recipient).is_zero() {
+                    let current_holders_count = self.pre_launch_holders_count.read();
 
-            if !self.launched.read() && self.balance_of(recipient) == 0 {
-                let current_holders_count = self.pre_launch_holders_count.read();
-                assert(
-                    current_holders_count < MAX_HOLDERS_BEFORE_LAUNCH, Errors::MAX_HOLDERS_REACHED
-                );
+                    // assert max holders limit is not reached
+                    assert(
+                        current_holders_count < MAX_HOLDERS_BEFORE_LAUNCH,
```

Comment:
Souldn't it be `current_holders_count <= MAX_HOLDERS_BEFORE_LAUNCH`?`

Reply:
`current_holders_count` is not increase yet, so it's `<` but we can also check that `current_holders_count + 1 <= MAX_HOLDERS_BEFORE_LAUNCH`